stl/inc/algorithm

```diff
@@ -6849,6 +7165,95 @@ _FwdIt3 set_symmetric_difference(_ExPo&&, _FwdIt1 _First1, _FwdIt1 _Last1, _FwdI
     _REQUIRE_PARALLEL_ITERATOR(_FwdIt3);
     return _STD set_symmetric_difference(_First1, _Last1, _First2, _Last2, _Dest);
 }
+
+#ifdef __cpp_lib_concepts
+namespace ranges {
+    // ALIAS TEMPLATE set_symmetric_difference_result
+    template <class _In1, class _In2, class _Out>
+    using set_symmetric_difference_result = in_in_out_result<_In1, _In2, _Out>;
+
+    // VARIABLE ranges::set_symmetric_difference
+    class _Set_symmetric_difference_fn : private _Not_quite_object {
+    public:
+        using _Not_quite_object::_Not_quite_object;
+
+        // clang-format off
+        template <input_iterator _It1, sentinel_for<_It1> _Se1, input_iterator _It2, sentinel_for<_It2> _Se2,
+            weakly_incrementable _Out, class _Pr = ranges::less, class _Pj1 = identity, class _Pj2 = identity>
+            requires mergeable<_It1, _It2, _Out, _Pr, _Pj1, _Pj2>
+        constexpr set_symmetric_difference_result<_It1, _It2, _Out> operator()(_It1 _First1, _Se1 _Last1, _It2 _First2,
+            _Se2 _Last2, _Out _Result, _Pr _Pred = {}, _Pj1 _Proj1 = {}, _Pj2 _Proj2 = {}) const {
+            _Adl_verify_range(_First1, _Last1);
+            _Adl_verify_range(_First2, _Last2);
+            auto _UResult = _Set_symmetric_difference_unchecked(_Get_unwrapped(_STD move(_First1)),
+                _Get_unwrapped(_STD move(_Last1)), _Get_unwrapped(_STD move(_First2)),
+                _Get_unwrapped(_STD move(_Last2)), _Get_unwrapped_unverified(_STD move(_Result)), _Pass_fn(_Pred),
+                _Pass_fn(_Proj1), _Pass_fn(_Proj2));
+            _Seek_wrapped(_First1, _STD move(_UResult.in1));

```

Comment:
I am wondering whether this one would be helpful (If yes than it is future work. No change requested):

```cpp
template<class _It1, class _It2, class _Out, class _UIt1, class _UIt2, class _UOut>
in_in_out_result<_It1, _It2, _Out> _Seek_wrapped(_It1& _In1, _It2& _In2, _UOut& _Out, in_in_out_result<_UIt1, _UIt2, _UOut>&& _UResult);
```
Same for `in_out_result `and `in_in_result` which occur often enough that it makes sense

Reply:
Probably so. I think we should add a work item to scan through the unwrapping and rewrapping to find opportunities for simplification after/in conjunction with the pass to unwrap result iterators with `_Get_unwrapped_n`.